### PR TITLE
feat(functions): nsip-document migration

### PR DESCRIPTION
**Pull request by JamesP-KC**
## Describe your changes

Migrator function and endpoint for `nsip-document` entity. 

Takes data from ODW `document_meta_data` table, transforms, and inserts it into the Back Office `Document` and `DocumentVersion` tables. 

The `folderId` for a Document is determined by the `path` attribute, and looked up in a Map, falling back to creating a new folder if one is not found in the Map.

## Issue ticket number and link

https://pins-ds.atlassian.net/browse/BOAS-1404

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/functions/applications-migration/nsip-document-migration/src/nsip-document-migration.js
Comment (beejjacobs) [inline apps/functions/applications-migration/nsip-document-migration/src/nsip-document-migration.js:1]: Can we move this to `common/migrators`, and restructure so there is a function to migrate documents for one case - which would get called by `migrate-case` - to match the others?

File: apps/functions/applications-migration/common/migrators/nsip-project-migration.js
Comment (beejjacobs) [inline apps/functions/applications-migration/common/migrators/nsip-project-migration.js:1]: Is this to support the new schema versions with nulls?

**Review by beejjacobs — CHANGES_REQUESTED** [review]
Generally looks good and I could follow it through.

We've got quite a few `console.log` statements, givens we're in the API context can we import logger instead and ensure any parameters are passed in an object? Then the messages will be JSON formatted and easier to query in app insights.

File: apps/functions/applications-migration/common/migrators/nsip-project-migration.js
Comment (JamesP-KC) [reply apps/functions/applications-migration/common/migrators/nsip-project-migration.js:1]: This is a bit of a stop-gap solution as those fields are all required in the schema, but are not included in the data from ODW, so have populated them manually with `null` values to get it through the validation.

File: apps/api/src/server/migration/migrators/nsip-document-migrator.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+import { getDocumentFolderId } from './folder/folder.js';
+import logger from '#utils/logger.js';
+
+/**
+ * Handle an HTTP trigger/request to run the migration
+ *
+ * @param {import('pins-data-model').Schemas.NSIPDocument[]} documents
+ */
+export const migrateNsipDocuments = async (documents) => {
+	console.info(`Migrating ${documents.length} documents`);
```
Comment (beejjacobs) [inline apps/api/src/server/migration/migrators/nsip-document-migrator.js:14]: Just missed one!

**Review by beejjacobs — APPROVED** [review]

**Pull request merged**